hawkular-alerts-rest/hawkular-alerts-rest-api/src/main/java/org/hawkular/alerts/rest/StreamWatcher.java

```diff
@@ -0,0 +1,306 @@
+import org.hawkular.alerts.api.services.EventsCriteria;
+import org.jboss.logging.Logger;
+
+/**
+ * Handle StreamingOutput logic for events/alerts Watchers.
+ *
+ * @author Jay Shaughnessy
+ * @author Lucas Ponce
+ */
+@Stateless
+public class StreamWatcher {
+    private static final Logger log = Logger.getLogger(StreamWatcher.class);
+    private static final Pager ctimePager;
+    private static final Pager stimePager;
+    private static final long WATCHER_INTERVAL_DEFAULT = 5 * 1000;
+    private static final long CLEAN_INTERVAL = 10 * 1000;
+    private static final long LEAP_INTERVAL = 1 * 1000;
+
+    @EJB
+    AlertsService alertsService;
+
+    static {
+        List<Order> ordering = new ArrayList<>();
+        ordering.add(Order.by("stime", Order.Direction.ASCENDING));
+        stimePager = new Pager(0, PageContext.UNLIMITED_PAGE_SIZE, ordering);
+        ordering = new ArrayList<>();
+        ordering.add(Order.by("ctime", Order.Direction.ASCENDING));
+        ctimePager = new Pager(0, PageContext.UNLIMITED_PAGE_SIZE, ordering);
+    }
+
+    public StreamingOutput watchAlerts(Set<String> tenantIds, AlertsCriteria criteria, Long watchInterval) {
+        return output -> {
+            Writer writer = new BufferedWriter(new OutputStreamWriter(output));
+            Long startWatchTime = criteria.getEndStatusTime();
+            Page<Alert> initialAlerts;
+            try {
+                initialAlerts = alertsService.getAlerts(tenantIds, criteria, stimePager);
+            } catch (Exception e) {
+                log.debug(e.getMessage(), e);
+                try {
+                    writer.write(JsonUtil.toJson(new ResponseUtil.ApiError(e.getMessage())) + "\r\n");
+                    writer.flush();
+                } catch (IOException io) {
+                    log.debug("Watcher client disconnected");
+                    try {
+                        writer.close();
+                    } catch (Exception ignored) {}
+                }
+                return;
+            }
+            if (initialAlerts == null) {
+                return;
+            }
+            Set<WatchedId> watchedIds = new HashSet<>();
+            initialAlerts.stream().forEach(alert -> {
+                try {
+                    writer.write(JsonUtil.toJson(alert) + "\r\n");
+                    writer.flush();
+                } catch (IOException io) {
+                    log.debug("Watcher client disconnected");
+                    try {
+                        writer.close();
+                    } catch (Exception ignored) {}
+                    return;
+                }
+                watchedIds.add(new WatchedId(alert.getId(), alert.getCurrentLifecycle().getStime()));
+            });
+            startWatchTime = startWatchTime == null ? System.currentTimeMillis() : startWatchTime;
+            long sleepWatcher = watchInterval == null ? WATCHER_INTERVAL_DEFAULT : watchInterval * 1000;
```

Comment:
The reason I questioned WATCHER_INTERVAL_DEFAULT=5000 in the original comment is because I misread this line.  I intepreted it as:
`watchInterval == (null ? WATCHER_INTERVAL_DEFAULT : watchInterval) * 1000`
But actually it is OK, because it is parsed as:
`watchInterval == null ? WATCHER_INTERVAL_DEFAULT : (watchInterval * 1000)`

I think some parentheses would have been more clear, but this is minor.